src/ferui-components/tree-view/tree-node-component.ts

```diff
@@ -0,0 +1,233 @@
+import {
+  Component,
+  Input,
+  OnInit,
+  Output,
+  EventEmitter,
+  TemplateRef,
+  HostBinding,
+  ElementRef,
+  ViewChild,
+} from '@angular/core';
+import {
+  TreeViewEvent,
+  TreeViewEventType,
+  TreeNode,
+  TreeViewColorTheme,
+  TreeNodeDataRetriever,
+  PagedTreeNodeDataRetriever,
+} from './interfaces';
+
+@Component({
+  selector: 'fui-tree-node',
+  template: `
+    <div class="fui-node-tree" (click)="onSelected()" [ngClass]="{ 'node-tree-selected': node.selected }">
+      <div [style.padding-left.px]="calculatePadding()" class="node-tree" #nodetree>
+        <span *ngIf="hasChildren" class="icon-template" (click)="onExpand()">
+          <ng-container [ngTemplateOutlet]="getIconTemplate() ? getIconTemplate() : defaultIconTemplate"></ng-container>
+          <ng-template #defaultIconTemplate>
+            <clr-icon class="expand-icon" *ngIf="node.expanded" shape="fui-less"></clr-icon>
+            <clr-icon class="expand-icon" *ngIf="!node.expanded" shape="fui-add"></clr-icon>
+          </ng-template>
+        </span>
+        <span class="label">
+          <ng-container
+            [ngTemplateOutlet]="getNodeTemplate() ? getNodeTemplate() : defaultNodeRenderer"
+            [ngTemplateOutletContext]="{ node: node }"
+          ></ng-container>
+          <ng-template #defaultNodeRenderer let-node="node">
+            <span>{{ node.data.data[node.data.label] }}</span>
+          </ng-template>
+        </span>
+      </div>
+    </div>
+    <div [style.margin-left.px]="calculatePadding() + 20">
+      <clr-icon *ngIf="node.showLoader" class="fui-loader-animation" shape="fui-spinner"></clr-icon>
+      <clr-icon *ngIf="node.loadError" class="fui-error-icon" shape="fui-error" aria-hidden="true"></clr-icon>
+      <span *ngIf="node.loadError" class="error-msg">Couldn't load content</span>
+    </div>
+  `,
+  styles: [
```

Comment:
Please, do not use the style directly in the angular component. You should create the style in ferui-design. And you need to add a class to the component so that you can avoid using the :host. 
We need all components to have their style in ferui-design only.

Reply:
Added to ferui-design